src/k8s-engine/images/imageProcessor.ts

```diff
@@ -0,0 +1,269 @@
+    const results = data.trimEnd().split(/\r?\n/).slice(1).map((line) => {
+      const [imageName, tag, imageID, size] = line.split(/\s+/);
+
+      return {
+        imageName, tag, imageID, size
+      };
+    });
+
+    return results;
+  }
+
+  /**
+   * Takes the `childProcess` returned by a command like `child_process.spawn` and processes the
+   * output streams and exit code and signal.
+   *
+   * @param child
+   * @param subcommandName - used for error messages only
+   * @param sendNotifications
+   */
+  async processChildOutput(child: ChildProcess, subcommandName: string, sendNotifications: boolean): Promise<childResultType> {
+    const result = { stdout: '', stderr: '' };
+
+    return await new Promise((resolve, reject) => {
+      child.stdout?.on('data', (data: Buffer) => {
+        const dataString = data.toString();
+
+        if (sendNotifications) {
+          this.emit('image-process-output', dataString, false);
+        }
+        result.stdout += dataString;
+      });
+      child.stderr?.on('data', (data: Buffer) => {
+        const dataString = data.toString();
+
+        result.stderr += dataString;
+        if (sendNotifications) {
+          this.emit('image-process-output', dataString, true);
+        }
+      });
+      child.on('exit', (code, signal) => {
+        if (result.stderr) {
+          // nerdctl complains about sha256 images
+          result.stderr = result.stderr.replace(/time=".*?"\s+level=warning\s+msg="unparsable image name \\"\w+@sha256:\w+\\""\s+/g, '');
+        }
+        if (result.stderr) {
+          const timeLessMessage = result.stderr.replace(/\btime=".*?"/g, '');
+
+          if (this.lastErrorMessage !== timeLessMessage) {
+            this.lastErrorMessage = timeLessMessage;
+            this.sameErrorMessageCount = 1;
+            console.log(result.stderr.replace(/(?!<\r)\n/g, '\r\n'));
+          } else {
+            const m = /(Error: .*)/.exec(this.lastErrorMessage);
+
+            this.sameErrorMessageCount += 1;
+            console.log(`${ this.processorName } ${ subcommandName }: ${ m ? m[1] : 'same error message' } #${ this.sameErrorMessageCount }\r`);
+          }
+        }
+        if (code === 0) {
+          resolve({ ...result, code });
+        } else if (signal) {
+          reject({
+            ...result, code: -1, signal
+          });
+        } else {
+          reject({ ...result, code });
+        }
+      });
+    });
+  }
+
+  getNamespaces(): Promise<Array<string>> {
+    throw new Error(`getNamespaces: not implemented for class ${ this.processorName }`);
+  }
+
+  get namespace() {
+    return this.currentNamespace;
+  }
+
+  set namespace(value: string) {
+    this.currentNamespace = value;
+  }
+
+  /* Subclass-specific method stubs here: */
+
+  abstract isInstallValid(mgr: K8s.KubernetesBackend, endpoint?: string): Promise<boolean>;
+
+  abstract install(backend: K8s.KubernetesBackend, force: boolean, address?: string): void;
```

Comment:
Since docker, nerdctl, and podman don't need to install, I expect we'll be able to drop this when kim has been removed.